sdl2promela/promela/modelbuilder.py

```diff
@@ -0,0 +1,209 @@
+import typing
+from typing import List, Set, SupportsRound, Tuple, Type, Dict
+from enum import Enum
+
+from sdl2promela.promela.model import InlineParameter, Model, Inline, \
+    Alternative, Block, Break, Skip, Switch, Do, Statement, Call, VariableDeclaration, VariableReference, \
+    Assignment, BinaryExpression, BinaryOperator, BlockType
+
+class ModelBuilder:
+    model : Model
+
+    def __init__(self):
+        self.model = Model()
+        self.model.prologue = ""
+        self.model.epilogue = ""
+        self.model.inlines = []
+
+    def build(self) -> Model:
+        return self.model
+
+    def withPrologue(self, prologue : str) -> 'ModelBuilder':
+        self.model.prologue = prologue
+        return self
+
+    def withEpilogue(self, epilogue : str) -> 'ModelBuilder':
+        self.model.epilogue = epilogue
+        return self
+
+    def withInline(self, inline : Inline) -> 'ModelBuilder':
+        self.model.inlines.append(inline)
+        return self
+
+class InlineBuilder:
+    inline : Inline
+
+    def __init__(self):
+        self.inline = Inline()
+        self.inline.parameters = []
+
+    def build(self) -> Inline:
+        return self.inline
+
+    def withName(self, name : str) -> 'InlineBuilder':
+        self.inline.name = name
+        return self
+
+    def withDefinition(self, definition : Block) -> 'InlineBuilder':
+        self.inline.definition = definition
+        return self
+
+    def withParameter(self, parameterName : str) -> 'InlineBuilder':
+        parameter = InlineParameter()
+        parameter.name = parameterName
+        self.inline.parameters.append(parameter)
+        return self
+
+class StatementsBuilder:
+    statements : List[Statement]
+
+    def __init__(self):
+        self.statements = []
+
+    def build(self) -> List[Statement]:
+        return self.statements
+
+    def withStatement(self, statement : Statement) -> 'StatementsBuilder':
+       self.statements.append(statement)
+       return self
+
+    def withStatements(self, statements : List[Statement]) -> 'StatementsBuilder':
+       self.statements.extend(statements)
+       return self
+
+
+class BlockBuilder:
+    block : Block
+
+    def __init__(self, type : BlockType):
+        self.block = Block()
+        self.block.type = type
+        self.block.statements = []
```

Comment:
shouldn't that be handled by `Block` constructor?

Reply:
Fair point...